app/controllers/sessions_controller.rb

```diff
@@ -1,9 +1,26 @@
 class SessionsController < Devise::SessionsController
+  include ControllerConcern
+
+  before_action :load_secret_code, only: %i(create new)
+  before_action :challenge_flash_messages, only: %i(create new)
+
   def destroy
     UserSignOutEvent.create(user_id: warden.user.user_id)
     signed_out = (Devise.sign_out_all_scopes ? sign_out : sign_out(resource_name))
     set_flash_message! :notice, :signed_out if signed_out
     yield if block_given?
     respond_to_on_destroy
   end
+
+  def challenge_flash_messages
+    return false if session.to_h.dig('challenge_parameters', 'flash_message').nil?
```

Comment:
```suggestion
    return if session.to_h.dig('challenge_parameters', 'flash_message').nil?
```
and not sure why all of this is happening.. it's no longer a flash message if we're manually putting in the session and deleting again...